src/main/java/org/swisspush/reststorage/RedisStorage.java

```diff
@@ -252,29 +267,30 @@ public void recomposeLuaScript() {
 
         /**
          * Load the get script into redis and store the sha in the class member sha.
-         * @param redisCommand the redis command that should be executed, after the script is loaded.
+         *
+         * @param redisCommand     the redis command that should be executed, after the script is loaded.
          * @param executionCounter a counter to control recursion depth
          */
         public void loadLuaScript(final RedisCommand redisCommand, int executionCounter) {
             final int executionCounterIncr = ++executionCounter;
 
             // check first if the lua script already exists in the store
-            redisClient.scriptExists(this.sha, resultArray -> {
-                if(resultArray.failed()){
+            redisAPI.script(Arrays.asList("exists", this.sha), resultArray -> {
+                if (resultArray.failed()) {
                     log.error("Error checking whether lua script exists", resultArray.cause());
                     return;
                 }
-                Long exists = resultArray.result().getLong(0);
+                Long exists = resultArray.result().get(0).toLong();

```

Comment:
Can `resultArray.result().get(0)` return `null` ?

Reply:
We should get 1 or 0 in the success case. I think we do not use the null check.